app/uk/gov/hmrc/serviceconfigs/persistence/DeploymentConfigSnapshotRepository.scala

```diff
@@ -66,16 +80,136 @@ class DeploymentConfigSnapshotRepository @Inject()(
       .toFuture
       .map(_ => ())
 
+  def removeLatestFlagForNonDeletedSnapshotsInEnvironment(environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.environment", environment.asString),
+          equal("latest", true),
+          equal("deleted", false),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+  def removeLatestFlagForServiceInEnvironment(serviceName: String, environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.name", serviceName),
+          equal("deploymentConfig.environment", environment.asString),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+
   def populate(date: Instant): Future[Unit] = {
```

Comment:
The logic of this is more complicated that I expected, is it because of mongo optimisations?

I'd have expected something like

```scala
forEnvironment(environment) =
  for {
    deploymentConfigs <- deploymentConfigRepository.findAll(environment) // excluding deleted here
    latestSnapshots   <- latestSnapshotsInEnvironment(environment)
    deletedServices   =  deploymentConfigs.map(_.serviceName).diff(latestSnapshot.map(_.serviceNames))
    _                 <- deploymentConfigs.foldLeftM(())(insertLatest(environment)
    _                 <- deletedServices.foldLeftM(())(deleteService(environment))
  } yield ()
  
where insertLatest(environment)(config) = 
  <transactional>
  for {
    _ <- removeLatestFlag    
    _ <- insertWithLatestFlag
  } yield ()
  
and deleteService(environment)(serviceName) =
  resetContentAndAddDeletedFlag  
```